zebra/zebra_routemap.c

```diff
@@ -1058,6 +1084,52 @@ static struct route_map_rule_cmd route_match_ip_address_prefix_len_cmd = {
 	route_match_ip_address_prefix_len_compile,
 	route_match_ip_address_prefix_len_free};
 
+static route_map_result_t
+route_match_ipv6_address_prefix_len(void *rule, struct prefix *prefix,
+				  route_map_object_t type, void *object)
+{
+	u_int32_t *prefixlen = (u_int32_t *)rule;
+
+	if (type == RMAP_ZEBRA) {
+		return ((prefix->prefixlen == *prefixlen) ? RMAP_MATCH
+							  : RMAP_NOMATCH);
+	}
+	return RMAP_NOMATCH;
+}
+
+static void *route_match_ipv6_address_prefix_len_compile(const char *arg)
+{
+	u_int32_t *prefix_len;
+	char *endptr = NULL;
+	unsigned long tmpval;
+
+	/* prefix len value shoud be integer. */
+	if (!all_digit(arg))
+		return NULL;
+
+	errno = 0;
+	tmpval = strtoul(arg, &endptr, 10);
+	if (*endptr != '\0' || errno || tmpval > UINT32_MAX)
+		return NULL;
+
+	prefix_len = XMALLOC(MTYPE_ROUTE_MAP_COMPILED, sizeof(u_int32_t));
+
+	if (!prefix_len)
+		return prefix_len;
+
+	*prefix_len = tmpval;
+	return prefix_len;
+}
+
+static void route_match_ipv6_address_prefix_len_free(void *rule)
+{
+	XFREE(MTYPE_ROUTE_MAP_COMPILED, rule);
+}
+
```

Comment:
I wonder if we could remove the three functions above and reuse the same functions we use for IPv4 route-map rules to avoid code duplication.